scripts/artifacts.cake

```diff
@@ -4,54 +4,68 @@
 ///  Generate the scripts which target the OmniSharp binaries.
 /// </summary>
 /// <param name="outputRoot">The root folder where the publised (or installed) binaries are located</param>
-void CreateRunScript(string outputRoot, string scriptFolder)
-{
+void CreateRunScript(string outputRoot, string scriptFolder, string name)
+ {
     if (IsRunningOnWindows())
     {
-        var desktopScript =  System.IO.Path.Combine(scriptFolder, "OmniSharp.cmd");
-        var coreScript = System.IO.Path.Combine(scriptFolder, "OmniSharp.Core.cmd");
-        var omniSharpPath = System.IO.Path.Combine(System.IO.Path.GetFullPath(outputRoot), "{0}", "OmniSharp");
-        var content = new string[] {
-                "SETLOCAL",
-                "",
-                $"\"{omniSharpPath}\" %*"
-            };
-        if (System.IO.File.Exists(desktopScript))
-        {
-            System.IO.File.Delete(desktopScript);
-        }
-        content[2] = String.Format(content[2], "net451");
-        System.IO.File.WriteAllLines(desktopScript, content);
-        if (System.IO.File.Exists(coreScript))
-        {
-            System.IO.File.Delete(coreScript);
-        }
-        content[2] = String.Format(content[2], "netcoreapp1.0");
-        System.IO.File.WriteAllLines(coreScript, content);
+        WriteWindowsScript(outputRoot, scriptFolder, name);
     }
     else
     {
-        var desktopScript = System.IO.Path.Combine(scriptFolder, "OmniSharp");
-        var coreScript = System.IO.Path.Combine(scriptFolder, "OmniSharp.Core");
-        var omniSharpPath = System.IO.Path.Combine(System.IO.Path.GetFullPath(outputRoot), "{1}", "OmniSharp");
-        var content = new string[] {
-                "#!/bin/bash",
-                "",
-                $"{{0}} \"{omniSharpPath}{{2}}\" \"$@\""
-            };
-        if (System.IO.File.Exists(desktopScript))
-        {
-            System.IO.File.Delete(desktopScript);
-        }
-        content[2] = String.Format(content[2], "mono", "net451", ".exe");
-        System.IO.File.WriteAllLines(desktopScript, content);
-        Run("chmod", $"+x \"{desktopScript}\"");
-        if (System.IO.File.Exists(coreScript))
-        {
-            System.IO.File.Delete(coreScript);
-        }
-        content[2] = String.Format(content[2], "", "netcoreapp1.0", "");
-        System.IO.File.WriteAllLines(coreScript, content);
-        Run("chmod", $"+x \"{desktopScript}\"");
+        WriteUnixScript(outputRoot, scriptFolder, name);
+    }
+}
+
+void WriteWindowsScript(string outputRoot, string scriptFolder, string name)
+{
+    var desktopScript =  System.IO.Path.Combine(scriptFolder, $"{name}.cmd");
+    var coreScript = System.IO.Path.Combine(scriptFolder, $"{name}.Core.cmd");
+    var omniSharpPath = System.IO.Path.Combine(System.IO.Path.GetFullPath(outputRoot), "{0}", "OmniSharp");
```

Comment:
in this case it is, but it's the path to the omnisharp root project... I'm shifting it around.

The original reason I had added the name parameter was to handle the scenario where everything including the C# project system were plugins...  After some self reflection that seemed to complicate things more than it helped.